back/engines/commercial/report_builder/spec/acceptance/reports_spec.rb

```diff
@@ -0,0 +1,269 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+resource 'Reports' do
+  explanation 'Customizable reports'
+
+  header 'Content-Type', 'application/json'
+
+  # Shared parameter descriptions
+  name_param_desc = 'The name of the report.'
+  craftjs_jsonmultiloc_param_desc = 'The craftjs layout configuration, as a multiloc string.'
+
+  get 'web_api/v1/reports' do
+    let_it_be(:reports) { create_list(:report, 3) }
+
+    describe 'when authorized' do
+      before { admin_header_token }
+
+      example_request 'List all reports' do
+        assert_status 200
+        expect(response_ids).to match_array(reports.pluck(:id))
+
+        # Layouts are not included when getting a collection of records.
+        expect(json_response_body[:included]).to be_nil
+      end
+    end
+
+    include_examples 'not authorized to visitors'
+    include_examples 'not authorized to normal users'
+  end
+
+  get 'web_api/v1/reports/:id' do
+    let_it_be(:report) { create(:report) }
+    let(:layout) { report.layout }
+    let(:id) { report.id }
+
+    describe 'when authorized' do
+      before { admin_header_token }
+
+      example_request 'Get one report by id' do
+        assert_status 200
+
+        expect(response_data).to match(
+          id: report.id,
+          type: 'report',
+          attributes: { name: report.name, updated_at: report.updated_at.iso8601(3) },
+          relationships: { layout: { data: { id: layout.id, type: 'content_builder_layout' } } }
```

Comment:
Is there a type in the layout object that identifies it as a report that we can test for?

Reply:
Sorry, looks like that happens further down.